File: Datastructures.js
function quickSort(array){
    if (array.length<1){
        return array;
    }
    var middle=array[0];
    var left=[];
    var right=[];
    for (let i=1;i<array.length;i++){
        if(array[i]<middle){
            left.push(array[i]);
        }else{
            right.push(array[i]);
        }
    }
    return quickSort(left).concat(middle,quickSort(right));
}
var unsorted=[5,4,3,2,1];
var sorted =quickSort(unsorted);
console.log(sorted);

File: datastructures.py
def sort(array):
    length=len(array)
    if length<2:
        return array
    left=[]
    right=[]
    mid=array[0]
    for i in range(1,length):
        if array[i]<mid:
            left.append(array[i])
        else :
            right.append(array[i])
    leftsort= sort(left)
    rightsort=sort(right)
    return (sort(left)+[mid]+sort(right))

unsorted =[10,9,5,4,7,5,3,2,0]
sorted_array = sort(unsorted)
print(sorted_array)
